calc.js

```diff
@@ -0,0 +1,145 @@
+prevOutput.innerText = '';
+curOutput.innerText = '';
+let newPrevOutput = '';
+let newCurOutput = '0';
+let mathOperator = '';
+
+function updateDisplay() {
+  prevOutput.innerText = newPrevOutput;
+  curOutput.innerText = newCurOutput;
+}
+updateDisplay();
+
+function calculate() {
+  let result;
+  newPrevOutput = parseFloat(newPrevOutput);
+  newCurOutput = parseFloat(newCurOutput);
+  if (Number.isNaN(newPrevOutput) || Number.isNaN(newCurOutput)) return;
+  if (mathOperator === '/') {
+    if (newCurOutput === 0) {
+      result = 'Math Err!';
+    } else {
+      result = newPrevOutput / newCurOutput;
+    }
+  } else if (mathOperator === '*') {
+    result = newPrevOutput * newCurOutput;
+  } else if (mathOperator === '+') {
+    result = newPrevOutput + newCurOutput;
+  } else if (mathOperator === '-') {
+    result = newPrevOutput - newCurOutput;
+  } else {
+    return;
+  }
+  newCurOutput = result;
+  mathOperator = undefined;
+  newPrevOutput = '';
+}
+
+function appendNumber(number) {
+  if (newCurOutput === '0' && number === '0') return;
+  if (newCurOutput.includes('.') && number === '.') return;
+  if (newCurOutput === '0' && number >= '1' && number <= '9') {
+    newCurOutput = number.toString();
+  } else if ((newCurOutput === '' || newCurOutput === 'Math Err!') && number === '.') {
+    newCurOutput = `0${number.toString()}`;
+  } else if (newCurOutput === 'Math Err!' && number !== '') {
+    newCurOutput = number.toString();
+  } else if (newCurOutput !== '' && number) {
+    newCurOutput += number.toString();
+  } else {
+    newCurOutput = newCurOutput.toString() + number.toString();
+  }
+}
+
+function chooseOperator(operator) {
+  if ((newCurOutput === 'Math Err!' || newCurOutput === '') && operator) return;
+  if (newPrevOutput.includes('Math Err!')) {
+    newCurOutput = newPrevOutput.slice(0, -3);
+    newPrevOutput = '';
+    return;
+  }
+  if (newCurOutput === '0' && operator) {
+    newPrevOutput = `${newCurOutput} ${operator} `;
+  }
+  if (newPrevOutput !== '') calculate();
+  mathOperator = operator;
+  newPrevOutput = `${newCurOutput} ${operator} `;
+  newCurOutput = '0';
+}
+
+function clearAll() {
+  while (prevOutput.innerText === '' && curOutput.innerText === '0') return;
+  prevOutput.innerText = '';
+  curOutput.innerText = '0';
+  newPrevOutput = '';
+  newCurOutput = '0';
+  mathOperator = '';
+}
+
+function deleteNum() {
+  if (newCurOutput === 'Math Err!') return;
+  if ((newCurOutput === '0' && newPrevOutput === '') || (newPrevOutput === '' && newCurOutput === '')) return;
+  if (newPrevOutput !== '' && (newCurOutput === '' || newCurOutput === '0')) {
+    newPrevOutput = newPrevOutput.slice(0, -3);
+    newCurOutput = newPrevOutput;
+    newPrevOutput = '';
+  } else {
+    newCurOutput = newCurOutput.toString().slice(0, -1);
+  }
+  updateDisplay();
+}
+
+function keyboardInput(e) {
+  if (e.key === '/' || e.key === '*' || e.key === '-' || e.key === '+') chooseOperator(e.key);
```

Comment:
- [ ] You could also allow the `x` keyboard button to call the multiply operator. While you could add this to your `logical or` statement, another way to write it would be:
```
const operators = ['/', 'x', '*', '+', '-'];

if (operators.includes(e.key) {
     .....
}
```

Reply:
Thanks boss! I'd update it now.